src/Angor/Client/Pages/Invest.razor

```diff
@@ -1272,38 +1207,58 @@ else
         return false;
     }
 
-    private void ShowChatModal()
+    private void ShowMessageModal()
     {
         showChatModal = true;
-        showNsec = false;
         var investorProject = project as InvestorProject;
         npubValue = NostrHelper.ConvertHexToNpub(investorProject?.InvestorNPub ?? "") ?? investorProject?.InvestorNPub ?? "";
 
         // Get founder's npub
-        founderNpub = NostrHelper.ConvertHexToNpub(investorProject.ProjectInfo.NostrPubKey) ??
-                      investorProject.ProjectInfo.NostrPubKey;
+        founderNpub = NostrHelper.ConvertHexToNpub(project.ProjectInfo.NostrPubKey) ??
+                      project.ProjectInfo.NostrPubKey;
+        
+        // If password is available, get the private key
+        if (passwordComponent.HasPassword())
```

Comment:
We normally separate this part to its own method like `ShowMessageModalAndCheckPassword` to trigger the password and then `ShowMessageModal`